### PR TITLE
FFI, I/O, polling & error handling cleanup

### PR DESCRIPTION
**Pull request by NiLuJe**
* Style: Use double-quotes
* Rewrite all the socket-related I/O methods to handle error and edge cases sanely.
* Rewrote the polling loop to be more easily readable.
* Fixed `WpaClient:readAllEvents` to prevent it from poll'ing and recv'ing more than once, as this was causing irritating slow-downs, especially in error cases.
* Unified error handling/passing.
* Switched to actual polling when waiting for a reply, instead of fixed sleep.
* Wait for a reply after *every* command, because sending a new command when wpa_supplicant hasn't processed the previous one will... screw things up.
* Random other cleanups & fixes.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/lj-wpaclient/8)

### CODE REVIEW

**Comment by NiLuJe** [comment]
I'll clear up the debugging tomorrow ;).

**Comment by Frenzie** [comment]
Is there anything we should test?

**Comment by NiLuJe** [comment]
Hopefully that it still works ;p.

So far, so good over here, both with system APs and KO APs.

**Comment by NiLuJe** [comment]
This should be ready now :).

File: ffi-cdecl/wpa_ctrl.cdef.c [hunk 1]
```diff
@@ -1,19 +1,27 @@
+#include <errno.h>
 #include <unistd.h>
 #include <sys/un.h>
 #include <sys/socket.h>
 #include <poll.h>
+#include <sys/select.h>
```
Comment (Frenzie) [inline ffi-cdecl/wpa_ctrl.cdef.c:6]: Is there a specific order in mind?

**Review by Frenzie — APPROVED** [review]
lgtm

File: ffi-cdecl/wpa_ctrl.cdef.c (hunk 1 above)
Comment (NiLuJe) [reply ffi-cdecl/wpa_ctrl.cdef.c:6]: None at all ^^. If this were actual C, I'd let clang-format sort it for me, FWIW ;).

Comment (Frenzie) [reply ffi-cdecl/wpa_ctrl.cdef.c:6]: In any case, I think it'd be clearer to keep it vaguely alphabetical and such. ;-)

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 4 of 9 files at r1, 1 of 4 files at r2, 1 of 3 files at r4, 2 of 2 files at r6, 1 of 1 files at r7, all commit messages.
> Dismissed @Frenzie from a discussion.
> *[Reviewable](https://reviewable.io/reviews/koreader/lj-wpaclient/8)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

___
*[ffi-cdecl/wpa_ctrl.cdef.c, line 6 at r5](https://reviewable.io/reviews/koreader/lj-wpaclient/8#-MnIiNm0obXLXGHPGXOF-r5-6:-MnRbRW66wAcVwEBqjSX:b-896fix) ([raw file](https://github.com/koreader/lj-wpaclient/blob/cf927e28d1b993ee279ad9d8be190af2cad12a11/ffi-cdecl/wpa_ctrl.cdef.c#L6)):*
<details><summary><i>Previously, Frenzie (Frans de Jonge) wrote…</i></summary><blockquote>

In any case, I think it'd be clearer to keep it vaguely alphabetical and such. ;-)
</blockquote></details>

Done.

**Pull request merged**